src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -0,0 +1,197 @@
+   * clockwise orientation relative their face's unit normal
+   * @param block A rock block
+   * @return A mapping from each face of the block to a Seq of vertices for that face arranged in counter-
+   *         clockwise order relative to its unit normal
+   */
+  private def orientVertices(block: Block): Map[Face, Seq[(Double, Double, Double)]] = {
+    val faces = block.findVertices
+    faces.keys.zip(
+      faces.map { kv =>
+        // Rotate vertices to all be in x-y plane
+        val R = BlockVTK.rotationMatrix(kv._1.normalVec, (0.0, 0.0, 1.0))
+        val rotatedVerts = kv._2 map { vertex =>
+          val rotatedVertex = R * DenseVector[Double](vertex._1, vertex._2, vertex._3)
+          (rotatedVertex(0), rotatedVertex(1), rotatedVertex(2))
+        }
+        // Order vertices in counter-clockwise orientation
+        val center = BlockVTK.findCenter(rotatedVerts)
+        val orderedVerts = {
+          if (kv._1.normalVec._3 < -BlockVTK.EPSILON) {
+            rotatedVerts.sortWith(BlockVTK.ccwCompare(_, _, center)).reverse
+          }
+          else {
+            rotatedVerts.sortWith(BlockVTK.ccwCompare(_, _, center))
+          }
+        }
+        // Rotate vertices back to original orientation
+        val invR = R.t // Inverse of rotation matrix is equal to its transpose
+        orderedVerts map { vertex =>
+          val orderedVertex = invR * DenseVector[Double](vertex._1, vertex._2, vertex._3)
+          (orderedVertex(0), orderedVertex(1), orderedVertex(2))
+        }
+      }
+    ).toMap
+  }
+
+  /**
+   * Finds the indices of the face vertices in the global list of vertices
+   * @param faceOrientedVerts Mapping from each face to a Seq of vertices for that face
+   * @param vertices Seq of distinct vertices for the rock block
+   * @return A mapping from each face of the block to a Seq of integers that represent
+   *         the indices of the vertices in the global vertex list
+   */
+  private def connectivity(faceOrientedVerts: Map[Face, Seq[(Double, Double, Double)]],
+                           vertices: Seq[(Double, Double, Double)]): Map[Face, Seq[Int]] = {
+    faceOrientedVerts.keys.zip(
+      faceOrientedVerts map { kv =>
+        kv._2 map { case vertex =>
+          vertices.indexOf(vertex)
+        }
+      }
+    ).toMap
+  }
+
+  /**
+   * Creates a sequence of all the normals of all the faces for the list of input blocks
+   * @param faces A mapping from each face of the block to a Seq of vertices for that face arranged in counter-
+   *         clockwise order relative to its unit normal
+   * @return Sequence of tuples representing normals of all the block faces
+   */
+  private def normals(faces: Map[Face, Seq[(Double, Double, Double)]]): Seq[(Double, Double, Double)] = {
+    faces map { kv =>
+      (kv._1.a, kv._1.b, kv._1.c)
+    }
+  }.toSeq
+
+  /**
+   * Determines the cumulative offsets that define each face in the connectivity list
```

Comment:
Perhaps is should just be offset? What I'm trying to say is that it finds the offset for each face's connectivities in the connectivity list based on how many vertices it has and also all the number of vertices for faces that precede it in the connectivity list.